node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/provider/ComponentsProviderImpl.java

```diff
@@ -89,30 +79,8 @@ public NodeAdminStateUpdater getNodeAdminStateUpdater() {
     }
 
     @Override
-    public MetricReceiverWrapper getMetricReceiverWrapper() {
-        return metricReceiverWrapper;
-    }
-
-
-    private void setCorePattern(Docker docker) {
-        final String[] sysctlCorePattern = {"sysctl", "-w", "kernel.core_pattern=" +
-                                            getDefaults().underVespaHome("var/crash/%e.core.%p")};
-        docker.executeInContainerAsRoot(NODE_ADMIN_CONTAINER_NAME, sysctlCorePattern);
-    }
-
-    private void initializeNodeAgentSecretAgent(Docker docker) {
-        final Path yamasAgentFolder = Paths.get("/etc/yamas-agent/");
-        docker.executeInContainerAsRoot(NODE_ADMIN_CONTAINER_NAME, "chmod", "a+w", yamasAgentFolder.toString());
-
-        Path nodeAdminCheckPath = Paths.get("/usr/bin/curl");
-        SecretAgentScheduleMaker scheduleMaker = new SecretAgentScheduleMaker("node-admin", 60, nodeAdminCheckPath,
-                "localhost:4080/rest/metrics");
-
-        try {
-            scheduleMaker.writeTo(yamasAgentFolder);
-            docker.executeInContainerAsRoot(NODE_ADMIN_CONTAINER_NAME, "service", "yamas-agent", "restart");
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to write secret-agent schedules for node-admin", e);
-        }
+    public void deconstruct() {
+        nodeAdminStateUpdater.stop();
```

Comment:
Here would be the natural place to log start of deconstruct and stop of deconstruct. Remove corresponding from nodeadminstateupdater.

Reply:
Fixed.